src/Pact/Native.hs

```diff
@@ -547,29 +550,54 @@ identity :: RNativeFun e
 identity _ [a] = return a
 identity i as = argsError i as
 
-stringHashNativeFun :: RNativeFun e
-stringHashNativeFun i as =
+hash' :: RNativeFun e
+hash' i as =
   case as of
     [TLitString s] -> go $ encodeUtf8 s
     [a] -> go $ toStrict $ encode a
     _ -> argsError i as
   where go = return . tStr . asString . hash
 
-hexStringToIntegerFun :: RNativeFun e
-hexStringToIntegerFun i as =
+strToInt :: RNativeFun e
+strToInt i as =
   case as of
-    [TLitString s] ->
-      bool (argsError i as) (go . T.take 128 $ s) $ -- truncate at 128 chars
-        T.all isHexDigit s
-    _ -> argsError i as 
+    [TLitString s] -> go 10 s
+    [TLitInteger base, TLitString s] -> go base s
+    _ -> argsError i as
   where
```

Comment:
Rather than repeat the body of `go`, I think separating the clauses of the overloads from the actual work makes it much clearer what is happening in each case - that is, if we do not have a `base`, then we defer to decimal, and if not, we follow through with the supplied base.